--- a/app/src/main/java/de/danoeh/antennapod/fragment/QueueFragment.java
+++ b/app/src/main/java/de/danoeh/antennapod/fragment/QueueFragment.java
@@ -515,6 +515,9 @@
             txtvEmpty.setVisibility(View.VISIBLE);
         } else {
             txtvEmpty.setVisibility(View.GONE);
+            if (recyclerView.getVisibility() == View.GONE) {
+                recyclerAdapter.notifyDataSetChanged();
+            }
             recyclerView.setVisibility(View.VISIBLE);
         }
 