--- a/app/src/main/java/fr/neamar/kiss/forwarder/ExperienceTweaks.java
+++ b/app/src/main/java/fr/neamar/kiss/forwarder/ExperienceTweaks.java
@@ -77,6 +77,8 @@
                 if(!prefs.getBoolean("double-tap", false)) {
                     if (prefs.getBoolean("history-onclick", false)) {
                         doAction("display-history");
+                    } else if (isMinimalisticModeEnabledForFavorites()) {
+                        mainActivity.favoritesBar.setVisibility(View.VISIBLE);
                     }
                 }
                 return super.onSingleTapUp(e);
@@ -88,6 +90,8 @@
                 if(prefs.getBoolean("double-tap", false)) {
                     if (prefs.getBoolean("history-onclick", false)) {
                         doAction("display-history");
+                    } else if (isMinimalisticModeEnabledForFavorites()) {
+                        mainActivity.favoritesBar.setVisibility(View.VISIBLE);
                     }
                 }
 