diff --git a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/RoutineCreateOrEditFragment.java b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/RoutineCreateOrEditFragment.java
index 38a44729..3c2e8722 100644
--- a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/RoutineCreateOrEditFragment.java
+++ b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/RoutineCreateOrEditFragment.java
@@ -119,7 +119,7 @@ public class RoutineCreateOrEditFragment extends DialogFragment implements Radia
                     timePickerFragment.show(getActivity().getSupportFragmentManager(), "TIME_PICKER_FRAGMENT");
                 } else {
                     TimePickerBuilder tpb = new TimePickerBuilder()
-                            .setFragmentManager(getChildFragmentManager())
+                            .setFragmentManager(getActivity().getSupportFragmentManager())
                             .setStyleResId(R.style.BetterPickersDialogFragment_Light);
                     tpb.addTimePickerDialogHandler(RoutineCreateOrEditFragment.this);
                     tpb.show();
