diff --git a/app/src/main/java/com/beemdevelopment/aegis/ui/slides/CustomAuthenticatedSlide.java b/app/src/main/java/com/beemdevelopment/aegis/ui/slides/CustomAuthenticatedSlide.java
index 10e45695..640101f0 100644
--- a/app/src/main/java/com/beemdevelopment/aegis/ui/slides/CustomAuthenticatedSlide.java
+++ b/app/src/main/java/com/beemdevelopment/aegis/ui/slides/CustomAuthenticatedSlide.java
@@ -45,11 +45,12 @@ public class CustomAuthenticatedSlide extends Fragment implements ISlidePolicy,
         _checkPasswordVisibility.setOnCheckedChangeListener((buttonView, isChecked) -> {
             if (isChecked) {
                 _textPassword.setTransformationMethod(null);
+                _textPasswordConfirm.setTransformationMethod(null);
                 _textPassword.clearFocus();
-                _textPasswordConfirm.setEnabled(false);
+                _textPasswordConfirm.clearFocus();
             } else {
                 _textPassword.setTransformationMethod(new PasswordTransformationMethod());
-                _textPasswordConfirm.setEnabled(true);
+                _textPasswordConfirm.setTransformationMethod(new PasswordTransformationMethod());
             }
         });