diff --git a/GPSTest/src/main/java/com/android/gpstest/Preferences.java b/GPSTest/src/main/java/com/android/gpstest/Preferences.java
index 2be69704..e1dbb3de 100644
--- a/GPSTest/src/main/java/com/android/gpstest/Preferences.java
+++ b/GPSTest/src/main/java/com/android/gpstest/Preferences.java
@@ -135,11 +135,8 @@ public class Preferences extends PreferenceActivity {
         chkDarkTheme.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
             @Override
             public boolean onPreferenceChange(Preference preference, Object newValue) {
-                // Close and restart Activities (clearing the back stack) to apply new theme
-                finish();
-                final Intent intent = getIntent();
-                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
-                startActivity(intent);
+                // Destroy and recreate Activity
+                recreate();
                 return true;
             }
         });
