diff --git a/app/src/main/java/org/shadowice/flocke/andotp/Preferences/PasswordEncryptedPreference.java b/app/src/main/java/org/shadowice/flocke/andotp/Preferences/PasswordEncryptedPreference.java
index 7302f8d3..6e089649 100644
--- a/app/src/main/java/org/shadowice/flocke/andotp/Preferences/PasswordEncryptedPreference.java
+++ b/app/src/main/java/org/shadowice/flocke/andotp/Preferences/PasswordEncryptedPreference.java
@@ -135,8 +136,7 @@ public class PasswordEncryptedPreference extends DialogPreference
             passwordConfirm.setInputType(InputType.TYPE_CLASS_NUMBER | InputType.TYPE_NUMBER_VARIATION_PASSWORD);
         }
 
-        passwordInput.setTransformationMethod(new PasswordTransformationMethod());
-        passwordConfirm.setTransformationMethod(new PasswordTransformationMethod());
+        ConfirmedPasswordTransformationHelper.setup(passwordLayout, passwordInput, passwordConfirm);
 
         passwordConfirm.addTextChangedListener(this);
         passwordInput.addTextChangedListener(this);
