diff --git a/app/src/main/java/de/danoeh/antennapod/fragment/AllEpisodesFragment.java b/app/src/main/java/de/danoeh/antennapod/fragment/AllEpisodesFragment.java
index ff7af97f6..5997d227a 100644
--- a/app/src/main/java/de/danoeh/antennapod/fragment/AllEpisodesFragment.java
+++ b/app/src/main/java/de/danoeh/antennapod/fragment/AllEpisodesFragment.java
@@ -341,11 +344,20 @@
     private void onFragmentLoaded() {
-        if (listAdapter == null) {
-            MainActivity mainActivity = (MainActivity) getActivity();
-            listAdapter = new AllEpisodesRecycleAdapter(mainActivity, itemAccess,
-                    new DefaultActionButtonCallback(mainActivity), showOnlyNewEpisodes());
-            listAdapter.setHasStableIds(true);
-            recyclerView.setAdapter(listAdapter);
+        if (episodes != null && episodes.size() > 0) {
+            if (listAdapter == null) {
+                MainActivity mainActivity = (MainActivity) getActivity();
+                listAdapter = new AllEpisodesRecycleAdapter(mainActivity, itemAccess,
+                        new DefaultActionButtonCallback(mainActivity), showOnlyNewEpisodes());
+                listAdapter.setHasStableIds(true);
+                recyclerView.setAdapter(listAdapter);
+            }
+            emptyView.setVisibility(View.GONE);
+            recyclerView.setVisibility(View.VISIBLE);
+        } else {
+            listAdapter = null;
+            recyclerView.setVisibility(View.GONE);
+            emptyView.setVisibility(View.VISIBLE);
         }
+
         listAdapter.notifyDataSetChanged();
         restoreScrollPosition();
         getActivity().supportInvalidateOptionsMenu();