diff --git a/app/src/main/java/io/pslab/activity/RoboticArmActivity.java b/app/src/main/java/io/pslab/activity/RoboticArmActivity.java
index 8c44102f..64a4ad48 100644
--- a/app/src/main/java/io/pslab/activity/RoboticArmActivity.java
+++ b/app/src/main/java/io/pslab/activity/RoboticArmActivity.java
@@ -300,11 +300,11 @@ public class RoboticArmActivity extends GuideActivity {
         seekArc4.setOnSeekArcChangeListener(new SeekArc.OnSeekArcChangeListener() {
             @Override
-            public void onProgressChanged(SeekArc seekArc, int i, boolean b) {
+            public void onProgressChanged(SeekArc seekArc, float progress, boolean b) {
                 if (editEnter) {
                     degreeText4.setText(String.valueOf(degree));
                     editEnter = false;
                 } else {
-                    degreeText4.setText(String.valueOf((int) (i * 3.6)));
+                    degreeText4.setText(String.valueOf(Math.round(progress * 3.6f)));
                 }
                 degreeText4.setCursorVisible(false);
             }