diff --git a/app/src/main/java/com/german_software_engineers/trainerapp/ExerciseViewActivity.java b/app/src/main/java/com/german_software_engineers/trainerapp/ExerciseView/Activity/ExerciseViewActivity.java
index cd5e171..aab808a 100644
--- a/app/src/main/java/com/german_software_engineers/trainerapp/ExerciseViewActivity.java
+++ b/app/src/main/java/com/german_software_engineers/trainerapp/ExerciseView/Activity/ExerciseViewActivity.java
@@ -59,10 +59,10 @@
     protected void onStart() {
         super.onStart();
         TextView ScheduleInfo = findViewById(R.id.ScheduleInfo);
 
         fragment = ExcersizeListFragment.newInstance(1, ActiveSchedule.getName(), Controller);
         getSupportFragmentManager().beginTransaction().replace(R.id.execView, fragment).commit();
 
-        String scheduleInfo = String.format(getResources().getString(R.string.ScheduleInfo), ActiveSchedule.getRepetitions(), ActiveSchedule.getPauseTime(), ActiveSchedule.getSets(), ActiveSchedule.getSpeed());
+        String scheduleInfo = String.format(getResources().getString(R.string.ScheduleInfo), ActiveSchedule.getRepetitions(), ActiveSchedule.getPauseTime(), ActiveSchedule.getSets(), ActiveSchedule.getSpeed(), ActiveSchedule.getTrainingsType().toString());
         ScheduleInfo.setText(scheduleInfo);
     }