diff --git a/app/src/main/java/com/fieldbook/tracker/activities/ConfigActivity.java b/app/src/main/java/com/fieldbook/tracker/activities/ConfigActivity.java
index 96178113..8f9fec4d 100644
--- a/app/src/main/java/com/fieldbook/tracker/activities/ConfigActivity.java
+++ b/app/src/main/java/com/fieldbook/tracker/activities/ConfigActivity.java
@@ -231,8 +231,11 @@
                     case 2:
                         if (!ep.getBoolean("ImportFieldFinished", false)) {
                             Utils.makeToast(getApplicationContext(),getString(R.string.warning_field_missing));
                             return;
+                        } else if (dt.getTraitColumnsAsString() == null) {
+                            Utils.makeToast(getApplicationContext(),getString(R.string.warning_traits_missing));
+                            return;
                         }
                         break;