--- a/app/src/main/java/com/beemdevelopment/aegis/ui/Dialogs.java
+++ b/app/src/main/java/com/beemdevelopment/aegis/ui/Dialogs.java
@@ -89,6 +89,7 @@
         EditText textPassword = view.findViewById(R.id.text_password);
         EditText textPasswordConfirm = view.findViewById(R.id.text_password_confirm);
         CheckBox switchToggleVisibility = view.findViewById(R.id.check_toggle_visibility);
+        final AtomicReference<Button> buttonOK = new AtomicReference<>();
 
         switchToggleVisibility.setOnCheckedChangeListener((CompoundButton.OnCheckedChangeListener) (buttonView, isChecked) -> {
             if (isChecked) {
@@ -99,6 +100,10 @@
                 textPassword.setTransformationMethod(new PasswordTransformationMethod());
                 textPasswordConfirm.setEnabled(true);
             }
+
+            // update the OK button state when the checkbox is toggled
+            boolean equal = EditTextHelper.areEditTextsEqual(textPassword, textPasswordConfirm);
+            buttonOK.get().setEnabled(equal || isChecked);
         });
 
         AlertDialog dialog = new AlertDialog.Builder(activity)
@@ -108,7 +113,6 @@
                 .setNegativeButton(android.R.string.cancel, null)
                 .create();
 
-        final AtomicReference<Button> buttonOK = new AtomicReference<>();
         dialog.setOnShowListener(d -> {
             Button button = dialog.getButton(AlertDialog.BUTTON_POSITIVE);
             button.setEnabled(false);