--- a/app/src/main/java/io/pslab/activity/RoboticArmActivity.java
+++ b/app/src/main/java/io/pslab/activity/RoboticArmActivity.java
@@ -479,11 +479,10 @@
         }
     }
     private void processDegreeInput(EditText degreeText, SeekArc seekArc) {
-        int previousDegree = degree;
         String degreeStr = degreeText.getText().toString().trim();
 
         if (degreeStr.isEmpty()) {
-            degree = (degree == 0) ? (int) (seekArc.getProgress() * 3.6) : previousDegree;
+            degree = (int) (seekArc.getProgress() * 3.6);
         } else {
             try {
                 degree = Integer.parseInt(degreeStr);