diff --git a/app/src/main/java/com/simon/harmonichackernews/ComposeActivity.java b/app/src/main/java/com/simon/harmonichackernews/ComposeActivity.java
index 835cb6f..3152071 100644
--- a/app/src/main/java/com/simon/harmonichackernews/ComposeActivity.java
+++ b/app/src/main/java/com/simon/harmonichackernews/ComposeActivity.java
@@ -210,12 +208,12 @@ public class ComposeActivity extends AppCompatActivity {
                 //we just do the default back behavior (which is predictive back)
                 AlertDialog dialog = new MaterialAlertDialogBuilder(editText.getContext())
                         .setMessage(type == TYPE_POST ? "Discard post?" : "Discard comment?")
-                        .setNegativeButton("Discard", new DialogInterface.OnClickListener() {
+                        .setPositiveButton("Discard", new DialogInterface.OnClickListener() {
                             public void onClick(DialogInterface dialog, int whichButton) {
                                 dialog.cancel();
                                 finish();
                             }})
-                        .setPositiveButton("Cancel", null).create();
+                        .setNegativeButton("Cancel", null).create();
 
                 dialog.show();
             }
