diff --git a/AnglersLog/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java b/android/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
index 6f6b3bf4..6dee7d91 100644
--- a/AnglersLog/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
+++ b/android/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
@@ -218,9 +222,9 @@ public class ManageLocationFragment extends ManageContentFragment {
 
         // get all possible fishing spots to be used when adding a new spot
         // this is purely for user convenience
-        List<UserDefineObject> allPossibleFishingSpots = mFishingSpots;
-        mFishingSpots.addAll(getNewLocation().getFishingSpots());
-        fragment.setFishingSpots(allPossibleFishingSpots);
+        Set<UserDefineObject> allPossibleFishingSpots = new HashSet<>(mFishingSpots);
+        allPossibleFishingSpots.addAll(getNewLocation().getFishingSpots());
+        fragment.setFishingSpots(new ArrayList<>(allPossibleFishingSpots));
 
         fragment.setOnVerifyInterface(new ManageFishingSpotFragment.OnVerifyInterface() {
             @Override

diff --git a/AnglersLog/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java b/android/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
index 6f6b3bf4..6dee7d91 100644
--- a/AnglersLog/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
+++ b/android/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
@@ -190,8 +189,7 @@ public class ManageLocationFragment extends ManageContentFragment {
         fishingSpotView.setOnClickDetailButton(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                mFishingSpots.remove(spot);
-                mContainer.removeView(fishingSpotView);
+                removeFishingSpot(spot, fishingSpotView);
             }
         });

diff --git a/AnglersLog/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java b/android/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
index 6f6b3bf4..6dee7d91 100644
--- a/AnglersLog/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
+++ b/android/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
@@ -47,7 +48,7 @@ public class ManageLocationFragment extends ManageContentFragment {
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.fragment_manage_location, container, false);
 
-        mContainer = (LinearLayout)view.findViewById(R.id.container);
+        mFishingSpotContainer = (LinearLayout)view.findViewById(R.id.container);
 
         initNameView(view);
         initAddFishingSpotView(view);

diff --git a/AnglersLog/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java b/android/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
index 6f6b3bf4..6dee7d91 100644
--- a/AnglersLog/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
+++ b/android/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
@@ -153,11 +155,8 @@ public class ManageLocationFragment extends ManageContentFragment {
 
     private void updateAllFishingSpots() {
         // remove all old views
-        for (MoreDetailView fishingSpotView : mFishingSpotViews) {
-            ViewGroup parent = ((ViewGroup)fishingSpotView.getParent());
-            if (parent != null)
-                parent.removeView(fishingSpotView);
-        }
+        mFishingSpotContainer.removeAllViews();
+        mFishingSpotViews.clear();
 
         for (UserDefineObject spot : mFishingSpots)
             addFishingSpot((FishingSpot)spot);
