diff --git a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/RoutineCreateOrEditFragment.java b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/RoutineCreateOrEditFragment.java
index 38a44729..3c2e8722 100644
--- a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/RoutineCreateOrEditFragment.java
+++ b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/RoutineCreateOrEditFragment.java
@@ -119,7 +119,7 @@ public class RoutineCreateOrEditFragment extends DialogFragment implements Radia
                     timePickerFragment.show(getActivity().getSupportFragmentManager(), "TIME_PICKER_FRAGMENT");
                 } else {
                     TimePickerBuilder tpb = new TimePickerBuilder()
-                            .setFragmentManager(getChildFragmentManager())
+                            .setFragmentManager(getActivity().getSupportFragmentManager())
                             .setStyleResId(R.style.BetterPickersDialogFragment_Light);
                     tpb.addTimePickerDialogHandler(RoutineCreateOrEditFragment.this);
                     tpb.show();

diff --git a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
index 3369cf9e..420cfd16 100644
--- a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
+++ b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
@@ -311,7 +311,7 @@ public class ScheduleTimetableFragment extends Fragment
                         new RadialTimePickerDialogFragment()
                                 .setOnTimeSetListener(ScheduleTimetableFragment.this)
                                 .setStartTime(time.getHourOfDay(), time.getMinuteOfHour());
-                timePickerDialog.show(getChildFragmentManager(), "111");
+                timePickerDialog.show(getActivity().getSupportFragmentManager(), "111");
             }
         });

diff --git a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
index 3369cf9e..420cfd16 100644
--- a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
+++ b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
@@ -539,7 +539,7 @@ public class ScheduleTimetableFragment extends Fragment
                         .setMinNumber(BigDecimal.ONE)
                         .setMaxNumber(BigDecimal.valueOf(31))
                         .setPlusMinusVisibility(NumberPicker.INVISIBLE)
-                        .setFragmentManager(getChildFragmentManager())
+                        .setFragmentManager(getActivity().getSupportFragmentManager())
                         .setTargetFragment(this).setReference(REF_DIALOG_ROUTINE_INTERVAL)
                         .setStyleResId(R.style.BetterPickersDialogFragment_Calendula);
         npb.show();

diff --git a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
index 3369cf9e..420cfd16 100644
--- a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
+++ b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
@@ -592,7 +592,7 @@ public class ScheduleTimetableFragment extends Fragment
 
         dialog.setArguments(b);
         dialog.setOnRecurrenceSetListener(this);
-        dialog.show(getChildFragmentManager(), "REC");
+        dialog.show(getActivity().getSupportFragmentManager(), "REC");
     }
 
     void addTimetableEntries(int timesPerDay, List<Routine> routines) {

diff --git a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/MedicineCreateOrEditFragment.java b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/MedicineCreateOrEditFragment.java
index a6822c2f..50eac905 100644
--- a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/MedicineCreateOrEditFragment.java
+++ b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/MedicineCreateOrEditFragment.java
@@ -483,7 +483,7 @@ public class MedicineCreateOrEditFragment extends Fragment implements SharedPref
                         //.setLabelText(ref == DIALOG_STOCK_ADD ? "Increase stock by" : "Decrease stock by")
                         .setDecimalVisibility(NumberPicker.VISIBLE)
                         .setPlusMinusVisibility(NumberPicker.INVISIBLE)
-                        .setFragmentManager(getChildFragmentManager())
+                        .setFragmentManager(getActivity().getSupportFragmentManager())
                         .setTargetFragment(this).setReference(ref)
                         .setStyleResId(R.style.BetterPickersDialogFragment_Calendula);
         npb.show();

diff --git a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
index 3369cf9e..420cfd16 100644
--- a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
+++ b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
@@ -1018,7 +1018,7 @@ public class ScheduleTimetableFragment extends Fragment
                                 .setMinNumber(BigDecimal.ONE)
                                 .setMaxNumber(BigDecimal.valueOf(100D))
                                 .setPlusMinusVisibility(NumberPicker.INVISIBLE)
-                                .setFragmentManager(getChildFragmentManager())
+                                .setFragmentManager(getActivity().getSupportFragmentManager())
                                 .setTargetFragment(ScheduleTimetableFragment.this)
                                 .setReference(REF_DIALOG_CYCLE_DAYS)
                                 .setStyleResId(R.style.BetterPickersDialogFragment_Calendula);
@@ -1034,7 +1034,7 @@ public class ScheduleTimetableFragment extends Fragment
                                 .setMinNumber(BigDecimal.ONE)
                                 .setMaxNumber(BigDecimal.valueOf(100D))
                                 .setPlusMinusVisibility(NumberPicker.INVISIBLE)
-                                .setFragmentManager(getChildFragmentManager())
+                                .setFragmentManager(getActivity().getSupportFragmentManager())
                                 .setTargetFragment(ScheduleTimetableFragment.this)
                                 .setReference(REF_DIALOG_CYCLE_REST)
                                 .setStyleResId(R.style.BetterPickersDialogFragment_Calendula);
