diff --git a/gpslogger/src/main/java/com/mendhak/gpslogger/GpsLoggingService.java b/gpslogger/src/main/java/com/mendhak/gpslogger/GpsLoggingService.java
index 5581ac7f..be5e5614 100644
--- a/gpslogger/src/main/java/com/mendhak/gpslogger/GpsLoggingService.java
+++ b/gpslogger/src/main/java/com/mendhak/gpslogger/GpsLoggingService.java
@@ -1074,6 +1074,7 @@ public class GpsLoggingService extends Service  {
         }
         catch(Exception e){
              LOG.error(getString(R.string.could_not_write_to_file), e);
+             Systems.showErrorNotification(this, getString(R.string.could_not_write_to_file));
         }
 
         session.clearDescription();
@@ -1174,6 +1175,14 @@ public class GpsLoggingService extends Service  {
     }
 
 
+    @EventBusHook
+    public void onEvent(CommandEvents.FileWriteFailure writeFailure){
+        Systems.showErrorNotification(this, getString(R.string.could_not_write_to_file));
+        if(writeFailure.stopLoggingDueToNMEA){
+            LOG.error("Could not write to NMEA file, stopping logging due to high frequency of write failures");
+            stopLogging();
+        }
+    }
 
     @EventBusHook
     public void onEvent(ProfileEvents.SwitchToProfile switchToProfileEvent){
diff --git a/gpslogger/src/main/java/com/mendhak/gpslogger/common/events/CommandEvents.java b/gpslogger/src/main/java/com/mendhak/gpslogger/common/events/CommandEvents.java
index c38d2729..4e0ea73e 100644
--- a/gpslogger/src/main/java/com/mendhak/gpslogger/common/events/CommandEvents.java
+++ b/gpslogger/src/main/java/com/mendhak/gpslogger/common/events/CommandEvents.java
@@ -72,4 +72,20 @@ public class CommandEvents {
      */
     public static class LogOnce {
     }
+
+    /**
+     * Used to indicate that the file write failed.
+     * The intention is to then notify the user of the failure since it represents data loss.
+     * Pass stopLogging, if true, ask the logging to stop. Use this for NMEA which is very high frequency.
+     */
+    public static class FileWriteFailure {
+        public boolean stopLoggingDueToNMEA;
+
+        public FileWriteFailure(){
+            this.stopLoggingDueToNMEA = false;
+        }
+        public FileWriteFailure(boolean stopLoggingDueToNMEA) {
+            this.stopLoggingDueToNMEA = stopLoggingDueToNMEA;
+        }
+    }
 }
diff --git a/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/geojson/GeoJSONWriterPoints.java b/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/geojson/GeoJSONWriterPoints.java
index cd28eb0f..1dfac8ac 100644
--- a/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/geojson/GeoJSONWriterPoints.java
+++ b/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/geojson/GeoJSONWriterPoints.java
@@ -4,6 +4,7 @@ import android.location.Location;
 import androidx.annotation.NonNull;
 
 import com.mendhak.gpslogger.common.Strings;
+import com.mendhak.gpslogger.common.events.CommandEvents;
 import com.mendhak.gpslogger.common.slf4j.Logs;
 import com.mendhak.gpslogger.loggers.Files;
 
@@ -13,6 +14,8 @@ import java.io.File;
 import java.io.IOException;
 import java.io.RandomAccessFile;
 
+import de.greenrobot.event.EventBus;
+
 /**
  * Created by clemens on 10.05.17.
  */
@@ -66,8 +69,8 @@ public class GeoJSONWriterPoints implements Runnable {
                 raf.close();
             }
         } catch (IOException e) {
-            e.printStackTrace();
-            LOG.error("GeoJSONWriterPoints", e);
+            EventBus.getDefault().post(new CommandEvents.FileWriteFailure());
+            LOG.error("Failed to write to GeoJSON file", e);
         }
 
     }
diff --git a/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/gpx/Gpx10FileLogger.java b/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/gpx/Gpx10FileLogger.java
index 2cd6a145..36af142d 100644
--- a/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/gpx/Gpx10FileLogger.java
+++ b/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/gpx/Gpx10FileLogger.java
@@ -26,6 +26,7 @@ import com.mendhak.gpslogger.common.Maths;
 import com.mendhak.gpslogger.common.PreferenceHelper;
 import com.mendhak.gpslogger.common.RejectionHandler;
 import com.mendhak.gpslogger.common.Strings;
+import com.mendhak.gpslogger.common.events.CommandEvents;
 import com.mendhak.gpslogger.common.slf4j.Logs;
 import com.mendhak.gpslogger.loggers.FileLogger;
 import com.mendhak.gpslogger.loggers.Files;
@@ -37,6 +38,8 @@ import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 
+import de.greenrobot.event.EventBus;
+
 
 public class Gpx10FileLogger implements FileLogger {
     protected final static Object lock = new Object();
@@ -160,7 +163,8 @@ class Gpx10AnnotateHandler implements Runnable {
 
                 LOG.debug("Finished annotation to GPX10 File");
             } catch (Exception e) {
-                LOG.error("Gpx10FileLogger.annotate", e);
+                EventBus.getDefault().post(new CommandEvents.FileWriteFailure());
+                LOG.error("Error annotating GPX file", e);
             }
 
         }
@@ -241,7 +245,8 @@ class Gpx10WriteHandler implements Runnable {
                 LOG.debug("Finished writing to GPX10 file");
 
             } catch (Exception e) {
-                LOG.error("Gpx10FileLogger.write", e);
+                EventBus.getDefault().post(new CommandEvents.FileWriteFailure());
+                LOG.error("Error writing to GPX file", e);
             }
 
         }
diff --git a/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/kml/Kml22FileLogger.java b/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/kml/Kml22FileLogger.java
index 87e01242..b9952a83 100644
--- a/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/kml/Kml22FileLogger.java
+++ b/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/kml/Kml22FileLogger.java
@@ -24,6 +24,7 @@ import android.location.Location;
 import com.mendhak.gpslogger.common.PreferenceHelper;
 import com.mendhak.gpslogger.common.RejectionHandler;
 import com.mendhak.gpslogger.common.Strings;
+import com.mendhak.gpslogger.common.events.CommandEvents;
 import com.mendhak.gpslogger.common.slf4j.Logs;
 import com.mendhak.gpslogger.loggers.FileLogger;
 import com.mendhak.gpslogger.loggers.Files;
@@ -36,6 +37,8 @@ import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 
+import de.greenrobot.event.EventBus;
+
 public class Kml22FileLogger implements FileLogger {
     protected final static Object lock = new Object();
     private final static ThreadPoolExecutor EXECUTOR = new ThreadPoolExecutor(1, 1, 60, TimeUnit.SECONDS,
@@ -117,7 +120,8 @@ class Kml22AnnotateHandler implements Runnable {
 
             }
         } catch (Exception e) {
-            LOG.error("Kml22FileLogger.annotate", e);
+            EventBus.getDefault().post(new CommandEvents.FileWriteFailure());
+            LOG.error("Error writing KML annotation", e);
         }
     }
 
@@ -222,7 +226,8 @@ class Kml22WriteHandler implements Runnable {
             }
 
         } catch (Exception e) {
-            LOG.error("Kml22FileLogger.write", e);
+            EventBus.getDefault().post(new CommandEvents.FileWriteFailure());
+            LOG.error("Error writing KML file", e);
         }
     }
 }
diff --git a/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/nmea/NmeaFileLogger.java b/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/nmea/NmeaFileLogger.java
index bee4d5ad..5653fcfc 100644
--- a/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/nmea/NmeaFileLogger.java
+++ b/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/nmea/NmeaFileLogger.java
@@ -23,8 +23,12 @@ import com.mendhak.gpslogger.common.PreferenceHelper;
 import com.mendhak.gpslogger.common.RejectionHandler;
 import com.mendhak.gpslogger.common.Session;
 import com.mendhak.gpslogger.common.Strings;
+import com.mendhak.gpslogger.common.events.CommandEvents;
+import com.mendhak.gpslogger.common.slf4j.Logs;
 import com.mendhak.gpslogger.loggers.Files;
 
+import org.slf4j.Logger;
+
 import java.io.BufferedWriter;
 import java.io.File;
 import java.io.FileWriter;
@@ -33,9 +37,12 @@ import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 
+import de.greenrobot.event.EventBus;
+
 public class NmeaFileLogger {
 
     protected final static Object lock = new Object();
+    private static final Logger LOG = Logs.of(NmeaFileLogger.class);
     String fileName;
     private final static ThreadPoolExecutor EXECUTOR = new ThreadPoolExecutor(1, 1, 60, TimeUnit.SECONDS,
             new LinkedBlockingQueue<Runnable>(10), new RejectionHandler());
@@ -60,7 +67,8 @@ public class NmeaFileLogger {
             try {
                 nmeaFile.createNewFile();
             } catch (IOException e) {
-
+                LOG.error("Error creating NMEA file", e);
+                EventBus.getDefault().post(new CommandEvents.FileWriteFailure(true));
             }
         }
 
@@ -71,6 +79,7 @@ public class NmeaFileLogger {
 
 class NmeaWriteHandler implements Runnable {
 
+    private static final Logger LOG = Logs.of(NmeaWriteHandler.class);
     File gpxFile;
     String nmeaSentence;
 
@@ -92,7 +101,8 @@ class NmeaWriteHandler implements Runnable {
                 Files.addToMediaDatabase(gpxFile, "text/plain");
 
             } catch (IOException e) {
-
+                LOG.error("Error writing NMEA sentence", e);
+                EventBus.getDefault().post(new CommandEvents.FileWriteFailure(true));
             }
         }