diff --git a/Calendula/build.gradle b/Calendula/build.gradle
index fb8b57d..f6c6890 100644
--- a/Calendula/build.gradle
+++ b/Calendula/build.gradle
@@ -162,7 +162,7 @@ android {
 
 tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile) {
     kotlinOptions {
-        allWarningsAsErrors = true
+        allWarningsAsErrors = false
     }
 }
 
diff --git a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/MedicineCreateOrEditFragment.java b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/MedicineCreateOrEditFragment.java
index a6822c2..ea1ab39 100644
--- a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/MedicineCreateOrEditFragment.java
+++ b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/MedicineCreateOrEditFragment.java
@@ -483,7 +483,7 @@ public class MedicineCreateOrEditFragment extends Fragment implements SharedPref
                         //.setLabelText(ref == DIALOG_STOCK_ADD ? "Increase stock by" : "Decrease stock by")
                         .setDecimalVisibility(NumberPicker.VISIBLE)
                         .setPlusMinusVisibility(NumberPicker.INVISIBLE)
-                        .setFragmentManager(getChildFragmentManager())
+                        .setFragmentManager(getFragmentManager())
                         .setTargetFragment(this).setReference(ref)
                         .setStyleResId(R.style.BetterPickersDialogFragment_Calendula);
         npb.show();
diff --git a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/RoutineCreateOrEditFragment.java b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/RoutineCreateOrEditFragment.java
index 38a4472..ab55577 100644
--- a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/RoutineCreateOrEditFragment.java
+++ b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/RoutineCreateOrEditFragment.java
@@ -116,10 +116,10 @@ public class RoutineCreateOrEditFragment extends DialogFragment implements Radia
                     RadialTimePickerDialogFragment timePickerFragment = new RadialTimePickerDialogFragment()
                             .setOnTimeSetListener(RoutineCreateOrEditFragment.this)
                             .setStartTime(hour, minute);
-                    timePickerFragment.show(getActivity().getSupportFragmentManager(), "TIME_PICKER_FRAGMENT");
+                    timePickerFragment.show(getFragmentManager(), "TIME_PICKER_FRAGMENT");
                 } else {
                     TimePickerBuilder tpb = new TimePickerBuilder()
-                            .setFragmentManager(getChildFragmentManager())
+                            .setFragmentManager(getFragmentManager())
                             .setStyleResId(R.style.BetterPickersDialogFragment_Light);
                     tpb.addTimePickerDialogHandler(RoutineCreateOrEditFragment.this);
                     tpb.show();
diff --git a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleImportFragment.java b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleImportFragment.java
index b36c624..5e4adf3 100644
--- a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleImportFragment.java
+++ b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleImportFragment.java
@@ -527,7 +527,7 @@ public class ScheduleImportFragment extends Fragment
                         .setMinNumber(BigDecimal.ONE)
                         .setMaxNumber(BigDecimal.valueOf(31))
                         .setPlusMinusVisibility(NumberPicker.INVISIBLE)
-                        .setFragmentManager(getChildFragmentManager())
+                        .setFragmentManager(getFragmentManager())
                         .setTargetFragment(this).setReference(REF_DIALOG_ROUTINE_INTERVAL)
                         .setStyleResId(R.style.BetterPickersDialogFragment_Calendula);
         npb.show();
@@ -567,7 +567,7 @@ public class ScheduleImportFragment extends Fragment
 
         dialog.setArguments(b);
         dialog.setOnRecurrenceSetListener(this);
-        dialog.show(getChildFragmentManager(), "REC");
+        dialog.show(getFragmentManager(), "REC");
     }
 
     void addTimetableEntries(int timesPerDay, List<Routine> routines) {
@@ -1053,7 +1053,7 @@ public class ScheduleImportFragment extends Fragment
                                 .setMinNumber(BigDecimal.ONE)
                                 .setMaxNumber(BigDecimal.valueOf(100))
                                 .setPlusMinusVisibility(NumberPicker.INVISIBLE)
-                                .setFragmentManager(getChildFragmentManager())
+                                .setFragmentManager(getFragmentManager())
                                 .setTargetFragment(ScheduleImportFragment.this)
                                 .setReference(REF_DIALOG_CYCLE_DAYS)
                                 .setStyleResId(R.style.BetterPickersDialogFragment_Calendula);
@@ -1069,7 +1069,7 @@ public class ScheduleImportFragment extends Fragment
                                 .setMinNumber(BigDecimal.ONE)
                                 .setMaxNumber(BigDecimal.valueOf(100))
                                 .setPlusMinusVisibility(NumberPicker.INVISIBLE)
-                                .setFragmentManager(getChildFragmentManager())
+                                .setFragmentManager(getFragmentManager())
                                 .setTargetFragment(ScheduleImportFragment.this)
                                 .setReference(REF_DIALOG_CYCLE_REST)
                                 .setStyleResId(R.style.BetterPickersDialogFragment_Calendula);
@@ -1102,7 +1102,7 @@ public class ScheduleImportFragment extends Fragment
                 RadialTimePickerDialogFragment timePickerDialog = new RadialTimePickerDialogFragment()
                         .setOnTimeSetListener(ScheduleImportFragment.this)
                         .setStartTime(time.getHourOfDay(), time.getMinuteOfHour());
-                timePickerDialog.show(getChildFragmentManager(), "111");
+                timePickerDialog.show(getFragmentManager(), "111");
             }
         });
 
diff --git a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
index 3369cf9..9f8f2e3 100644
--- a/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
+++ b/Calendula/src/main/java/es/usc/citius/servando/calendula/fragments/ScheduleTimetableFragment.java
@@ -311,7 +311,7 @@ public class ScheduleTimetableFragment extends Fragment
                         new RadialTimePickerDialogFragment()
                                 .setOnTimeSetListener(ScheduleTimetableFragment.this)
                                 .setStartTime(time.getHourOfDay(), time.getMinuteOfHour());
-                timePickerDialog.show(getChildFragmentManager(), "111");
+                timePickerDialog.show(getFragmentManager(), "111");
             }
         });
 
@@ -539,7 +539,7 @@ public class ScheduleTimetableFragment extends Fragment
                         .setMinNumber(BigDecimal.ONE)
                         .setMaxNumber(BigDecimal.valueOf(31))
                         .setPlusMinusVisibility(NumberPicker.INVISIBLE)
-                        .setFragmentManager(getChildFragmentManager())
+                        .setFragmentManager(getFragmentManager())
                         .setTargetFragment(this).setReference(REF_DIALOG_ROUTINE_INTERVAL)
                         .setStyleResId(R.style.BetterPickersDialogFragment_Calendula);
         npb.show();
@@ -551,7 +551,7 @@ public class ScheduleTimetableFragment extends Fragment
                 .setMinNumber(1)
                 .setMaxNumber(24)
                 .setPlusMinusVisibility(NumberPicker.INVISIBLE)
-                .setFragmentManager(getChildFragmentManager())
+                .setFragmentManager(getFragmentManager())
                 .setTargetFragment(this)
                 .setReference(REF_DIALOG_HOURLY_INTERVAL)
                 .setStyleResId(R.style.BetterPickersDialogFragment_Calendula);
@@ -592,7 +592,7 @@ public class ScheduleTimetableFragment extends Fragment
 
         dialog.setArguments(b);
         dialog.setOnRecurrenceSetListener(this);
-        dialog.show(getChildFragmentManager(), "REC");
+        dialog.show(getFragmentManager(), "REC");
     }
 
     void addTimetableEntries(int timesPerDay, List<Routine> routines) {
@@ -1018,7 +1018,7 @@ public class ScheduleTimetableFragment extends Fragment
                                 .setMinNumber(BigDecimal.ONE)
                                 .setMaxNumber(BigDecimal.valueOf(100D))
                                 .setPlusMinusVisibility(NumberPicker.INVISIBLE)
-                                .setFragmentManager(getChildFragmentManager())
+                                .setFragmentManager(getFragmentManager())
                                 .setTargetFragment(ScheduleTimetableFragment.this)
                                 .setReference(REF_DIALOG_CYCLE_DAYS)
                                 .setStyleResId(R.style.BetterPickersDialogFragment_Calendula);
@@ -1034,7 +1034,7 @@ public class ScheduleTimetableFragment extends Fragment
                                 .setMinNumber(BigDecimal.ONE)
                                 .setMaxNumber(BigDecimal.valueOf(100D))
                                 .setPlusMinusVisibility(NumberPicker.INVISIBLE)
-                                .setFragmentManager(getChildFragmentManager())
+                                .setFragmentManager(getFragmentManager())
                                 .setTargetFragment(ScheduleTimetableFragment.this)
                                 .setReference(REF_DIALOG_CYCLE_REST)
                                 .setStyleResId(R.style.BetterPickersDialogFragment_Calendula);
