diff --git a/app/src/main/java/io/lerk/lrkFM/adapter/FileArrayAdapter.java b/app/src/main/java/io/lerk/lrkFM/adapter/FileArrayAdapter.java
index 3499e3e..5a30516 100644
--- a/app/src/main/java/io/lerk/lrkFM/adapter/FileArrayAdapter.java
+++ b/app/src/main/java/io/lerk/lrkFM/adapter/FileArrayAdapter.java
@@ -15,6 +15,7 @@ import android.widget.TextView;
 import android.widget.Toast;
 
 import java.text.SimpleDateFormat;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Objects;
 
