diff --git a/GPSTest/src/main/java/com/android/gpstest/GpsTestActivity.java b/GPSTest/src/main/java/com/android/gpstest/GpsTestActivity.java
index 9db5860..7307033 100644
--- a/GPSTest/src/main/java/com/android/gpstest/GpsTestActivity.java
+++ b/GPSTest/src/main/java/com/android/gpstest/GpsTestActivity.java
@@ -186,11 +186,14 @@ public class GpsTestActivity extends AppCompatActivity
 
     private SensorManager mSensorManager;
 
+    private boolean mDarkTheme;
+
     /** Called when the activity is first created. */
     @Override
     public void onCreate(Bundle savedInstanceState) {
         // Set theme
-        if (Application.getPrefs().getBoolean(getString(R.string.pref_key_dark_theme), false)) {
+        mDarkTheme = Application.getPrefs().getBoolean(getString(R.string.pref_key_dark_theme), false);
+        if (mDarkTheme) {
             setTheme(R.style.AppTheme_Dark_NoActionBar);
         }
         requestWindowFeature(Window.FEATURE_INDETERMINATE_PROGRESS);
@@ -251,6 +254,12 @@ public class GpsTestActivity extends AppCompatActivity
     protected void onResume() {
         super.onResume();
 
+        boolean darkTheme = Application.getPrefs().getBoolean(getString(R.string.pref_key_dark_theme), false);
+        if (mDarkTheme != darkTheme) {
+            mDarkTheme = darkTheme;
+            recreate();
+        }
+
         addStatusListener();
 
         addOrientationSensorListener();
diff --git a/GPSTest/src/main/java/com/android/gpstest/HelpActivity.java b/GPSTest/src/main/java/com/android/gpstest/HelpActivity.java
index 232e8e5..4fde7ce 100644
--- a/GPSTest/src/main/java/com/android/gpstest/HelpActivity.java
+++ b/GPSTest/src/main/java/com/android/gpstest/HelpActivity.java
@@ -25,6 +25,9 @@ public class HelpActivity extends AppCompatActivity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
+        if (Application.getPrefs().getBoolean(getString(R.string.pref_key_dark_theme), false)) {
+            setTheme(R.style.AppTheme_Dark);
+        }
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_help);
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
diff --git a/GPSTest/src/main/java/com/android/gpstest/Preferences.java b/GPSTest/src/main/java/com/android/gpstest/Preferences.java
index 2be6970..9eaf1ff 100644
--- a/GPSTest/src/main/java/com/android/gpstest/Preferences.java
+++ b/GPSTest/src/main/java/com/android/gpstest/Preferences.java
@@ -17,6 +17,7 @@
 package com.android.gpstest;
 
 import android.content.Intent;
+import android.content.SharedPreferences;
 import android.net.Uri;
 import android.os.Bundle;
 import android.preference.CheckBoxPreference;
@@ -135,11 +136,11 @@ public class Preferences extends PreferenceActivity {
         chkDarkTheme.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
             @Override
             public boolean onPreferenceChange(Preference preference, Object newValue) {
-                // Close and restart Activities (clearing the back stack) to apply new theme
-                finish();
-                final Intent intent = getIntent();
-                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
-                startActivity(intent);
+                // Restart Preferences activity to apply new theme
+                Application.getPrefs().edit()
+                        .putBoolean(getString(R.string.pref_key_dark_theme), (Boolean) newValue)
+                        .commit();
+                recreate();
                 return true;
             }
         });
