diff --git a/AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.java b/AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.java
index c394559..556303a 100644
--- a/AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.java
@@ -429,19 +429,23 @@ public class Reviewer extends AbstractFlashcardViewer {
             }
         }
 
+        MenuItem undoItem = menu.findItem(R.id.action_undo);
         if (mShowWhiteboard && mWhiteboard != null && mWhiteboard.undoSize() > 0) {
             // Whiteboard undo queue non-empty. Switch the undo icon to a whiteboard specific one.
-            menu.findItem(R.id.action_undo).setIcon(R.drawable.ic_eraser_variant_white_24dp);
+            undoItem.setIcon(R.drawable.ic_eraser_variant_white_24dp);
+            undoItem.setEnabled(!getControlBlocked());
         } else if (mShowWhiteboard && mWhiteboard != null && mWhiteboard.isUndoModeActive()) {
             // Whiteboard undo queue empty, but user has added strokes to it for current card. Disable undo button.
-            menu.findItem(R.id.action_undo).setIcon(R.drawable.ic_eraser_variant_white_24dp);
-            menu.findItem(R.id.action_undo).setEnabled(false).getIcon().setAlpha(Themes.ALPHA_ICON_DISABLED_LIGHT);
+            undoItem.setIcon(R.drawable.ic_eraser_variant_white_24dp);
+            undoItem.setEnabled(false);
         } else if (colIsOpen() && getCol().undoAvailable()) {
-            menu.findItem(R.id.action_undo).setIcon(R.drawable.ic_undo_white_24dp);
+            undoItem.setIcon(R.drawable.ic_undo_white_24dp);
+            undoItem.setEnabled(!getControlBlocked());
         } else {
-            menu.findItem(R.id.action_undo).setIcon(R.drawable.ic_undo_white_24dp);
-            menu.findItem(R.id.action_undo).setEnabled(false).getIcon().setAlpha(Themes.ALPHA_ICON_DISABLED_LIGHT);
+            undoItem.setIcon(R.drawable.ic_undo_white_24dp);
+            undoItem.setEnabled(false);
         }
+        undoItem.getIcon().setAlpha(undoItem.isEnabled() ? Themes.ALPHA_ICON_ENABLED_LIGHT : Themes.ALPHA_ICON_DISABLED_LIGHT);
         if (mPrefWhiteboard) {
             // Configure the whiteboard related items in the action bar
             menu.findItem(R.id.action_enable_whiteboard).setTitle(R.string.disable_whiteboard);
