diff --git a/app/src/main/java/com/vrem/wifianalyzer/vendor/model/VendorDB.java b/app/src/main/java/com/vrem/wifianalyzer/vendor/model/VendorDB.java
index b57990e..94c39d3 100644
--- a/app/src/main/java/com/vrem/wifianalyzer/vendor/model/VendorDB.java
+++ b/app/src/main/java/com/vrem/wifianalyzer/vendor/model/VendorDB.java
@@ -43,8 +43,8 @@ class VendorDB implements VendorService {
 
     VendorDB(@NonNull Resources resources) {
         this.resources = resources;
-        this.vendors = new TreeMap<>();
-        this.macs = new TreeMap<>();
+        this.vendors = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+        this.macs = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
         this.loaded = false;
     }
 
@@ -123,7 +123,7 @@ class VendorDB implements VendorService {
 
         @Override
         public boolean evaluate(String object) {
-            return object.contains(filter);
+            return StringUtils.containsIgnoreCase(object, filter);
         }
     }
 
