--- a/app/src/main/java/net/gsantner/markor/util/DocumentIO.java
+++ b/app/src/main/java/net/gsantner/markor/util/DocumentIO.java
@@ -137,7 +137,9 @@
             document.setFile(documentInitial.getFile());
         }
 
-        if (!currentText.equals(documentInitial.getContent())) {
+        boolean hasContentChanges = !currentText.equals(documentInitial.getContent());
+        boolean targetFileMissing = !document.getFile().exists();
+        if (hasContentChanges || targetFileMissing) {
             document.forceAddNextChangeToHistory();
             document.setContent(currentText);
             ret = FileUtils.writeFile(document.getFile(), document.getContent());