--- a/app/src/main/java/io/pslab/activity/RoboticArmActivity.java
+++ b/app/src/main/java/io/pslab/activity/RoboticArmActivity.java
@@ -68,7 +68,6 @@
     private SeekArc seekArc1, seekArc2, seekArc3, seekArc4;
     private LinearLayout servo1TimeLine, servo2TimeLine, servo3TimeLine, servo4TimeLine;
     private int degree;
-    private boolean editEnter = false;
     private HorizontalScrollView scrollView;
     private CountDownTimer timeLine;
     private boolean isPlaying = false;
@@ -225,12 +224,7 @@
         seekArc1.setOnSeekArcChangeListener(new SeekArc.OnSeekArcChangeListener() {
             @Override
             public void onProgressChanged(SeekArc seekArc, int i, boolean b) {
-                if (editEnter) {
-                    degreeText1.setText(String.valueOf(degree));
-                    editEnter = false;
-                } else {
-                    degreeText1.setText(String.valueOf((int) (i * 3.6)));
-                }
+                degreeText1.setText(String.valueOf((int) (i * 3.6)));
                 degreeText1.setCursorVisible(false);
             }
 
@@ -248,12 +242,7 @@
         seekArc2.setOnSeekArcChangeListener(new SeekArc.OnSeekArcChangeListener() {
             @Override
             public void onProgressChanged(SeekArc seekArc, int i, boolean b) {
-                if (editEnter) {
-                    degreeText2.setText(String.valueOf(degree));
-                    editEnter = false;
-                } else {
-                    degreeText2.setText(String.valueOf((int) (i * 3.6)));
-                }
+                degreeText2.setText(String.valueOf((int) (i * 3.6)));
                 degreeText2.setCursorVisible(false);
             }
 
@@ -271,12 +260,7 @@
         seekArc3.setOnSeekArcChangeListener(new SeekArc.OnSeekArcChangeListener() {
             @Override
             public void onProgressChanged(SeekArc seekArc, int i, boolean b) {
-                if (editEnter) {
-                    degreeText3.setText(String.valueOf(degree));
-                    editEnter = false;
-                } else {
-                    degreeText3.setText(String.valueOf((int) (i * 3.6)));
-                }
+                degreeText3.setText(String.valueOf((int) (i * 3.6)));
                 degreeText3.setCursorVisible(false);
             }
 
@@ -294,12 +278,7 @@
         seekArc4.setOnSeekArcChangeListener(new SeekArc.OnSeekArcChangeListener() {
             @Override
             public void onProgressChanged(SeekArc seekArc, int i, boolean b) {
-                if (editEnter) {
-                    degreeText4.setText(String.valueOf(degree));
-                    editEnter = false;
-                } else {
-                    degreeText4.setText(String.valueOf((int) (i * 3.6)));
-                }
+                degreeText4.setText(String.valueOf((int) (i * 3.6)));
                 degreeText4.setCursorVisible(false);
             }
 
@@ -501,7 +480,8 @@
             toastInvalidValueMessage();
         } else {
             seekArc.setProgress((int) (degree / 3.6));
-            editEnter = true;
+            degreeText.setText(String.valueOf(degree));
+            degreeText.setCursorVisible(false);
         }
     }
 