--- a/AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.java
@@ -429,18 +429,25 @@
             }
         }
 
+        MenuItem undoItem = menu.findItem(R.id.action_undo);
         if (mShowWhiteboard && mWhiteboard != null && mWhiteboard.undoSize() > 0) {
             // Whiteboard undo queue non-empty. Switch the undo icon to a whiteboard specific one.
-            menu.findItem(R.id.action_undo).setIcon(R.drawable.ic_eraser_variant_white_24dp);
+            undoItem.setIcon(R.drawable.ic_eraser_variant_white_24dp);
+            undoItem.setEnabled(true);
+            undoItem.getIcon().setAlpha(255);
         } else if (mShowWhiteboard && mWhiteboard != null && mWhiteboard.isUndoModeActive()) {
             // Whiteboard undo queue empty, but user has added strokes to it for current card. Disable undo button.
-            menu.findItem(R.id.action_undo).setIcon(R.drawable.ic_eraser_variant_white_24dp);
-            menu.findItem(R.id.action_undo).setEnabled(false).getIcon().setAlpha(Themes.ALPHA_ICON_DISABLED_LIGHT);
+            undoItem.setIcon(R.drawable.ic_eraser_variant_white_24dp);
+            undoItem.setEnabled(false);
+            undoItem.getIcon().setAlpha(Themes.ALPHA_ICON_DISABLED_LIGHT);
         } else if (colIsOpen() && getCol().undoAvailable()) {
-            menu.findItem(R.id.action_undo).setIcon(R.drawable.ic_undo_white_24dp);
+            undoItem.setIcon(R.drawable.ic_undo_white_24dp);
+            undoItem.setEnabled(true);
+            undoItem.getIcon().setAlpha(255);
         } else {
-            menu.findItem(R.id.action_undo).setIcon(R.drawable.ic_undo_white_24dp);
-            menu.findItem(R.id.action_undo).setEnabled(false).getIcon().setAlpha(Themes.ALPHA_ICON_DISABLED_LIGHT);
+            undoItem.setIcon(R.drawable.ic_undo_white_24dp);
+            undoItem.setEnabled(false);
+            undoItem.getIcon().setAlpha(Themes.ALPHA_ICON_DISABLED_LIGHT);
         }
         if (mPrefWhiteboard) {
             // Configure the whiteboard related items in the action bar