--- a/app/src/main/java/de/danoeh/antennapod/adapter/AllEpisodesRecycleAdapter.java
+++ b/app/src/main/java/de/danoeh/antennapod/adapter/AllEpisodesRecycleAdapter.java
@@ -106,6 +106,9 @@
     public void onBindViewHolder(final Holder holder, int position) {
         final FeedItem item = itemAccess.getItem(position);
         if (item == null) return;
+        holder.itemView.setAlpha(1.0f);
+        holder.itemView.setTranslationX(0.0f);
+        holder.itemView.setTranslationY(0.0f);
         holder.itemView.setOnLongClickListener(v -> {
             this.position = holder.getAdapterPosition();
             return false;
@@ -270,6 +273,8 @@
         @Override
         public void onItemClear() {
             itemView.setAlpha(1.0f);
+            itemView.setTranslationX(0.0f);
+            itemView.setTranslationY(0.0f);
         }
 
         public FeedItem getFeedItem() { return item; }
--- a/app/src/main/java/de/danoeh/antennapod/adapter/QueueRecyclerAdapter.java
+++ b/app/src/main/java/de/danoeh/antennapod/adapter/QueueRecyclerAdapter.java
@@ -92,6 +92,9 @@
 
     public void onBindViewHolder(ViewHolder holder, int pos) {
         FeedItem item = itemAccess.getItem(pos);
+        holder.itemView.setAlpha(1.0f);
+        holder.itemView.setTranslationX(0.0f);
+        holder.itemView.setTranslationY(0.0f);
         holder.bind(item);
         holder.itemView.setOnLongClickListener(v -> {
             selectedItem = item;
@@ -200,6 +203,8 @@
         @Override
         public void onItemClear() {
             itemView.setAlpha(1.0f);
+            itemView.setTranslationX(0.0f);
+            itemView.setTranslationY(0.0f);
         }
 
         public void bind(FeedItem item) {