--- a/app/src/main/java/io/pslab/activity/RoboticArmActivity.java
+++ b/app/src/main/java/io/pslab/activity/RoboticArmActivity.java
@@ -500,8 +500,11 @@
             seekArc.setProgress(0);
             toastInvalidValueMessage();
         } else {
+            editEnter = true;
             seekArc.setProgress((int) (degree / 3.6));
-            editEnter = true;
+            degreeText.setText(String.valueOf(degree));
+            degreeText.setCursorVisible(false);
+            editEnter = false;
         }
     }