--- a/app/src/main/java/com/trianguloy/urlchecker/modules/companions/PatternCatalog.java
+++ b/app/src/main/java/com/trianguloy/urlchecker/modules/companions/PatternCatalog.java
@@ -55,17 +55,17 @@
 
                 // privacy redirections samples (see https://github.com/TrianguloY/URLCheck/discussions/122)
                 .put("Reddit ➔ Teddit", new JSONObject()
-                        .put("regex", "^https?://(?:[a-z0-9-]+\\.)*?reddit.com/(.*)")
+                        .put("regex", "^https?://(?:[a-z0-9-]+\\.)*?reddit\\.com/(.*)")
                         .put("replacement", "https://teddit.net/$1")
                         .put("enabled", "false")
                 )
                 .put("Twitter ➔ Nitter", new JSONObject()
-                        .put("regex", "^https?://(?:[a-z0-9-]+\\.)*?twitter.com/(.*)")
+                        .put("regex", "^https?://(?:[a-z0-9-]+\\.)*?twitter\\.com/(.*)")
                         .put("replacement", "https://nitter.net/$1")
                         .put("enabled", "false")
                 )
                 .put("Youtube ➔ Invidious", new JSONObject()
-                        .put("regex", "^https?://(?:[a-z0-9-]+\\.)*?youtube.com/(.*)")
+                        .put("regex", "^https?://(?:[a-z0-9-]+\\.)*?youtube\\.com/(.*)")
                         .put("replacement", new JSONArray()
                                 .put("https://yewtu.be/$1")
                                 .put("https://farside.link/invidious/$1")