--- a/app/src/main/java/org/sufficientlysecure/htmltextview/LocalLinkMovementMethod.java
+++ b/app/src/main/java/org/sufficientlysecure/htmltextview/LocalLinkMovementMethod.java
@@ -23,6 +23,7 @@
 import android.text.method.Touch;
 import android.text.style.ClickableSpan;
 import android.view.MotionEvent;
+import android.view.ViewConfiguration;
 import android.widget.TextView;
 
 /**
@@ -30,6 +31,7 @@
  */
 public class LocalLinkMovementMethod extends LinkMovementMethod {
     static LocalLinkMovementMethod sInstance;
+    private long lastActionDownTime;
 
     public static LocalLinkMovementMethod getInstance() {
         if (sInstance == null)
@@ -61,8 +63,12 @@
 
             if (link.length != 0) {
                 if (action == MotionEvent.ACTION_UP) {
-                    link[0].onClick(widget);
+                    long elapsedTime = event.getEventTime() - lastActionDownTime;
+                    if (elapsedTime < ViewConfiguration.getLongPressTimeout()) {
+                        link[0].onClick(widget);
+                    }
                 } else if (action == MotionEvent.ACTION_DOWN) {
+                    lastActionDownTime = event.getEventTime();
                     Selection.setSelection(buffer,
                             buffer.getSpanStart(link[0]),
                             buffer.getSpanEnd(link[0]));