--- a/app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java
+++ b/app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java
@@ -226,12 +226,12 @@
     private Document loadDocument() {
         AppSettings appSettings = new AppSettings(getActivity().getApplicationContext());
         Document document = DocumentIO.loadDocument(getActivity(), getArguments(), _document);
-        if (document != null){
-            document.setDoHistory(appSettings.isEditorHistoryEnabled());
-        }
         if (document.getHistory().isEmpty()) {
             document.forceAddNextChangeToHistory();
             document.addToHistory();
+        }
+        if (document != null){
+            document.setDoHistory(appSettings.isEditorHistoryEnabled());
         }
 
         return document;
--- a/app/src/main/res/layout/document__fragment__edit.xml
+++ b/app/src/main/res/layout/document__fragment__edit.xml
@@ -46,7 +46,7 @@
             android:fontFamily="sans-serif-light"
             android:gravity="top"
             android:imeOptions="flagNoExtractUi"
-            android:inputType="textMultiLine|textCapSentences"
+            android:inputType="textMultiLine"
             android:paddingBottom="120dp"
             android:paddingLeft="@dimen/activity_horizontal_margin"
             android:paddingRight="@dimen/activity_horizontal_margin"