--- a/AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java
@@ -126,6 +126,7 @@
     public static final String INTERVAL = "interval";
     public static final String LAPSES = "lapses";
     public static final String NOTE = "note";
+    public static final String NOTE_ID = "noteId";
     public static final String REVIEWS = "reviews";
     private static Pattern sMarkedPattern = Pattern.compile(".*[Mm]arked.*");
 
@@ -389,7 +390,7 @@
         Integer[] checkedPositions = mCheckedCardPositions.toArray(new Integer[0]);
         HashSet<String> notes = new HashSet<>();
         for (Integer position : checkedPositions) {
-            String noteId = mCards.get(position).get(NOTE);
+            String noteId = mCards.get(position).get(NOTE_ID);
             if (notes.add(noteId) && notes.size() > 1) {
                 return false;
             }
@@ -1523,6 +1524,7 @@
         }
         item.put(LAPSES, Integer.toString(c.getLapses()));
         item.put(NOTE, c.model().optString("name"));
+        item.put(NOTE_ID, Long.toString(note.getId()));
         item.put(QUESTION, formatQA(q, context));
         item.put(REVIEWS, Integer.toString(c.getReps()));
         String tags = note.stringTags();