--- a/app/src/main/java/fr/neamar/kiss/forwarder/ExperienceTweaks.java
+++ b/app/src/main/java/fr/neamar/kiss/forwarder/ExperienceTweaks.java
@@ -78,6 +78,9 @@
                     if (prefs.getBoolean("history-onclick", false)) {
                         doAction("display-history");
                     }
+                    if (isMinimalisticModeEnabledForFavorites()) {
+                        mainActivity.favoritesBar.setVisibility(View.VISIBLE);
+                    }
                 }
                 return super.onSingleTapUp(e);
             }
@@ -88,6 +91,9 @@
                 if(prefs.getBoolean("double-tap", false)) {
                     if (prefs.getBoolean("history-onclick", false)) {
                         doAction("display-history");
+                    }
+                    if (isMinimalisticModeEnabledForFavorites()) {
+                        mainActivity.favoritesBar.setVisibility(View.VISIBLE);
                     }
                 }
 