--- a/app/src/main/java/free/rm/skytube/gui/businessobjects/adapters/GridViewHolder.java
+++ b/app/src/main/java/free/rm/skytube/gui/businessobjects/adapters/GridViewHolder.java
@@ -257,9 +257,6 @@
 		actionHandler.updateSubscribeMenuItem(youTubeVideo.getChannelId(), menu);
 
 		popupMenu.setOnMenuItemClickListener(item -> {
-			if (actionHandler.handleChannelActions(context, youTubeVideo.getChannel(), item.getItemId())) {
-				return true;
-			}
 			switch(item.getItemId()) {
 				case R.id.menu_open_video_with:
 					compositeDisposable.add(youTubeVideo.playVideoExternally(context).subscribe());
@@ -304,7 +301,7 @@
 					}
 					return true;
 			}
-			return false;
+			return actionHandler.handleChannelActions(context, youTubeVideo.getChannel(), item.getItemId());
 		});
 		popupMenu.show();
 	}