diff --git a/app/src/main/java/io/pslab/activity/RoboticArmActivity.java b/app/src/main/java/io/pslab/activity/RoboticArmActivity.java
index 8c44102..7ff8727 100644
--- a/app/src/main/java/io/pslab/activity/RoboticArmActivity.java
+++ b/app/src/main/java/io/pslab/activity/RoboticArmActivity.java
@@ -68,7 +68,6 @@ public class RoboticArmActivity extends GuideActivity {
     private SeekArc seekArc1, seekArc2, seekArc3, seekArc4;
     private LinearLayout servo1TimeLine, servo2TimeLine, servo3TimeLine, servo4TimeLine;
     private int degree;
-    private boolean editEnter = false;
     private HorizontalScrollView scrollView;
     private CountDownTimer timeLine;
     private boolean isPlaying = false;
@@ -225,10 +224,7 @@ public class RoboticArmActivity extends GuideActivity {
         seekArc1.setOnSeekArcChangeListener(new SeekArc.OnSeekArcChangeListener() {
             @Override
             public void onProgressChanged(SeekArc seekArc, int i, boolean b) {
-                if (editEnter) {
-                    degreeText1.setText(String.valueOf(degree));
-                    editEnter = false;
-                } else {
+                if (b) {
                     degreeText1.setText(String.valueOf((int) (i * 3.6)));
                 }
                 degreeText1.setCursorVisible(false);
@@ -248,10 +244,7 @@ public class RoboticArmActivity extends GuideActivity {
         seekArc2.setOnSeekArcChangeListener(new SeekArc.OnSeekArcChangeListener() {
             @Override
             public void onProgressChanged(SeekArc seekArc, int i, boolean b) {
-                if (editEnter) {
-                    degreeText2.setText(String.valueOf(degree));
-                    editEnter = false;
-                } else {
+                if (b) {
                     degreeText2.setText(String.valueOf((int) (i * 3.6)));
                 }
                 degreeText2.setCursorVisible(false);
@@ -271,10 +264,7 @@ public class RoboticArmActivity extends GuideActivity {
         seekArc3.setOnSeekArcChangeListener(new SeekArc.OnSeekArcChangeListener() {
             @Override
             public void onProgressChanged(SeekArc seekArc, int i, boolean b) {
-                if (editEnter) {
-                    degreeText3.setText(String.valueOf(degree));
-                    editEnter = false;
-                } else {
+                if (b) {
                     degreeText3.setText(String.valueOf((int) (i * 3.6)));
                 }
                 degreeText3.setCursorVisible(false);
@@ -294,10 +284,7 @@ public class RoboticArmActivity extends GuideActivity {
         seekArc4.setOnSeekArcChangeListener(new SeekArc.OnSeekArcChangeListener() {
             @Override
             public void onProgressChanged(SeekArc seekArc, int i, boolean b) {
-                if (editEnter) {
-                    degreeText4.setText(String.valueOf(degree));
-                    editEnter = false;
-                } else {
+                if (b) {
                     degreeText4.setText(String.valueOf((int) (i * 3.6)));
                 }
                 degreeText4.setCursorVisible(false);
@@ -500,8 +487,8 @@ public class RoboticArmActivity extends GuideActivity {
             seekArc.setProgress(0);
             toastInvalidValueMessage();
         } else {
+            degreeText.setText(String.valueOf(degree));
             seekArc.setProgress((int) (degree / 3.6));
-            editEnter = true;
         }
     }
 
