diff --git a/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java b/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
index 6f6b3bf..9c4e3de 100644
--- a/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
+++ b/app/src/main/java/com/cohenadair/anglerslog/locations/ManageLocationFragment.java
@@ -216,11 +216,7 @@ public class ManageLocationFragment extends ManageContentFragment {
         if (editingId != null)
             fragment.setIsEditing(true, editingId);
 
-        // get all possible fishing spots to be used when adding a new spot
-        // this is purely for user convenience
-        List<UserDefineObject> allPossibleFishingSpots = mFishingSpots;
-        mFishingSpots.addAll(getNewLocation().getFishingSpots());
-        fragment.setFishingSpots(allPossibleFishingSpots);
+        fragment.setFishingSpots(mFishingSpots);
 
         fragment.setOnVerifyInterface(new ManageFishingSpotFragment.OnVerifyInterface() {
             @Override
