diff --git a/app/src/main/java/io/pslab/fragment/ESPFragment.java b/app/src/main/java/io/pslab/fragment/ESPFragment.java
index 3c13b32..d3bb9d6 100644
--- a/app/src/main/java/io/pslab/fragment/ESPFragment.java
+++ b/app/src/main/java/io/pslab/fragment/ESPFragment.java
@@ -100,13 +100,16 @@ public class ESPFragment extends DialogFragment {
                 Request request = new Request.Builder()
                         .url("http://" + espIPAddress)
                         .build();
-                Response response = client.newCall(request).execute();
-                if (response.code() == 200) {
-                    ScienceLabCommon.setIsWifiConnected(true);
-                    ScienceLabCommon.setEspBaseIP(espIPAddress);
+                try (Response response = client.newCall(request).execute()) {
+                    if (response.code() == 200) {
+                        ScienceLabCommon.setIsWifiConnected(true);
+                        ScienceLabCommon.setEspBaseIP(espIPAddress);
+                    }
+                    if (response.body() != null) {
+                        result = response.body().string();
+                    }
                 }
-                result = response.body().string();
-            } catch (IOException e) {
+            } catch (Exception e) {
                 e.printStackTrace();
             }
             return result;
