diff --git a/app/src/main/java/org/shadowice/flocke/andotp/Activities/IntroScreenActivity.java b/app/src/main/java/org/shadowice/flocke/andotp/Activities/IntroScreenActivity.java
index c1c445d6..31724ea9 100644
--- a/app/src/main/java/org/shadowice/flocke/andotp/Activities/IntroScreenActivity.java
+++ b/app/src/main/java/org/shadowice/flocke/andotp/Activities/IntroScreenActivity.java
@@ -355,59 +355,92 @@ public class IntroScreenActivity extends IntroActivity {
                     Constants.AuthMethod authMethod = selectionMapping.get(i);
 
                     if (authMethod == Constants.AuthMethod.PASSWORD) {
-                        credentialsLayout.setVisibility(View.VISIBLE);
+                        setupForPasswordInput();
+                    } else if (authMethod == Constants.AuthMethod.PIN) {
+                        setupForPinInput();
+                    } else {
+                        credentialsLayout.setVisibility(View.INVISIBLE);
+                        UIHelper.hideKeyboard(getIntroActivity(), root);
+                    }
 
-                        passwordLayout.setHint(getString(R.string.settings_hint_password));
-                        passwordConfirm.setHint(R.string.settings_hint_password_confirm);
+                    passwordInput.setText(null);
+                    passwordConfirm.setText(null);
 
-                        passwordInput.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_PASSWORD);
-                        passwordConfirm.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_PASSWORD);
+                    authWarnings.setVisibility(View.GONE);
 
-                        passwordInput.setTransformationMethod(new PasswordTransformationMethod());
-                        passwordConfirm.setTransformationMethod(new PasswordTransformationMethod());
+                    updateNavigation();
+                }
 
-                        minLength = Constants.AUTH_MIN_PASSWORD_LENGTH;
-                        lengthWarning = getString(R.string.settings_label_short_password, minLength);
-                        noPasswordWarning = getString(R.string.intro_slide3_warn_no_password);
-                        confirmPasswordWarning = getString(R.string.intro_slide3_warn_confirm_password);
+                private void setupForPasswordInput() {
+                    credentialsLayout.setVisibility(View.VISIBLE);
 
-                        if (getIntroActivity().getCurrentSlidePosition() == slidePos) {
-                            passwordInput.requestFocus();
-                            UIHelper.showKeyboard(getContext(), passwordInput);
-                        }
-                    } else if (authMethod == Constants.AuthMethod.PIN) {
-                        credentialsLayout.setVisibility(View.VISIBLE);
+                    passwordLayout.setHint(getString(R.string.settings_hint_password));
+                    passwordConfirm.setHint(R.string.settings_hint_password_confirm);
 
-                        passwordLayout.setHint(getString(R.string.settings_hint_pin));
-                        passwordConfirm.setHint(R.string.settings_hint_pin_confirm);
+                    passwordInput.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_PASSWORD);
+                    passwordConfirm.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_PASSWORD);
 
-                        passwordInput.setInputType(InputType.TYPE_CLASS_NUMBER | InputType.TYPE_NUMBER_VARIATION_PASSWORD);
-                        passwordConfirm.setInputType(InputType.TYPE_CLASS_NUMBER | InputType.TYPE_NUMBER_VARIATION_PASSWORD);
+                    setPasswordTransformationMethod();
 
-                        passwordInput.setTransformationMethod(new PasswordTransformationMethod());
-                        passwordConfirm.setTransformationMethod(new PasswordTransformationMethod());
+                    minLength = Constants.AUTH_MIN_PASSWORD_LENGTH;
+                    lengthWarning = getString(R.string.settings_label_short_password, minLength);
+                    noPasswordWarning = getString(R.string.intro_slide3_warn_no_password);
+                    confirmPasswordWarning = getString(R.string.intro_slide3_warn_confirm_password);
 
-                        minLength = Constants.AUTH_MIN_PIN_LENGTH;
-                        lengthWarning = getString(R.string.settings_label_short_pin, minLength);
-                        noPasswordWarning = getString(R.string.intro_slide3_warn_no_pin);
-                        confirmPasswordWarning = getString(R.string.intro_slide3_warn_confirm_pin);
+                    focusOnPasswordInput();
+                }
 
-                        if (getIntroActivity().getCurrentSlidePosition() == slidePos) {
-                            passwordInput.requestFocus();
-                            UIHelper.showKeyboard(getContext(), passwordInput);
+                private void setPasswordTransformationMethod() {
+                    passwordLayout.setEndIconOnClickListener(new View.OnClickListener() {
+                        @Override
+                        public void onClick(View v) {
+                            boolean wasShowingPassword = passwordInput.getTransformationMethod() instanceof PasswordTransformationMethod;
+                            // Dispatch password visibility change to both password and confirm inputs
+                            dispatchPasswordVisibilityChange(passwordInput, wasShowingPassword);
+                            dispatchPasswordVisibilityChange(passwordConfirm, wasShowingPassword);
+                            passwordLayout.refreshDrawableState();
                         }
+                    });
+                    passwordInput.setTransformationMethod(PasswordTransformationMethod.getInstance());
+                    passwordConfirm.setTransformationMethod(PasswordTransformationMethod.getInstance());
+                }
+
+                private void dispatchPasswordVisibilityChange(EditText editText, boolean wasShowingPassword) {
+                    final int selection = editText.getSelectionEnd();
+                    if (wasShowingPassword) {
+                        editText.setTransformationMethod(null);
                     } else {
-                        credentialsLayout.setVisibility(View.INVISIBLE);
+                        editText.setTransformationMethod(PasswordTransformationMethod.getInstance());
+                    }
+                    if (selection >= 0) {
+                        editText.setSelection(selection);
+                    }
+                }
 
-                        UIHelper.hideKeyboard(getIntroActivity(), root);
+                private void focusOnPasswordInput() {
+                    if (getIntroActivity().getCurrentSlidePosition() == slidePos) {
+                        passwordInput.requestFocus();
+                        UIHelper.showKeyboard(getContext(), passwordInput);
                     }
+                }
 
-                    passwordInput.setText(null);
-                    passwordConfirm.setText(null);
+                private void setupForPinInput() {
+                    credentialsLayout.setVisibility(View.VISIBLE);
 
-                    authWarnings.setVisibility(View.GONE);
+                    passwordLayout.setHint(getString(R.string.settings_hint_pin));
+                    passwordConfirm.setHint(R.string.settings_hint_pin_confirm);
 
-                    updateNavigation();
+                    passwordInput.setInputType(InputType.TYPE_CLASS_NUMBER | InputType.TYPE_NUMBER_VARIATION_PASSWORD);
+                    passwordConfirm.setInputType(InputType.TYPE_CLASS_NUMBER | InputType.TYPE_NUMBER_VARIATION_PASSWORD);
+
+                    setPasswordTransformationMethod();
+
+                    minLength = Constants.AUTH_MIN_PIN_LENGTH;
+                    lengthWarning = getString(R.string.settings_label_short_pin, minLength);
+                    noPasswordWarning = getString(R.string.intro_slide3_warn_no_pin);
+                    confirmPasswordWarning = getString(R.string.intro_slide3_warn_confirm_pin);
+
+                    focusOnPasswordInput();
                 }
 
                 @Override
diff --git a/app/src/main/res/layout/component_intro_authentication.xml b/app/src/main/res/layout/component_intro_authentication.xml
index ac18d641..1b709eec 100644
--- a/app/src/main/res/layout/component_intro_authentication.xml
+++ b/app/src/main/res/layout/component_intro_authentication.xml
@@ -64,7 +64,7 @@
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
                 android:hint="@string/settings_hint_password"
-                app:passwordToggleEnabled="true" >
+                app:endIconMode="password_toggle" >
 
                 <com.google.android.material.textfield.TextInputEditText
                     android:id="@+id/introPasswordEdit"