diff --git a/AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java b/AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java
index a8dba179ee..1810a94874 100644
--- a/AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java
@@ -43,6 +43,12 @@ public class Previewer extends AbstractFlashcardViewer {
 
         mCardList = getIntent().getLongArrayExtra("cardList");
         mIndex = getIntent().getIntExtra("index", -1);
+
+        if (savedInstanceState != null){
+            mIndex = savedInstanceState.getInt("index", mIndex);
+            mShowingAnswer = savedInstanceState.getBoolean("showingAnswer", mShowingAnswer);
+        }
+
         if (mCardList.length == 0 || mIndex < 0 || mIndex > mCardList.length - 1) {
             Timber.e("Previewer started with empty card list or invalid index");
             finishWithoutAnimation();
@@ -58,7 +64,12 @@ public class Previewer extends AbstractFlashcardViewer {
     protected void onCollectionLoaded(Collection col) {
         super.onCollectionLoaded(col);
         mCurrentCard = col.getCard(mCardList[mIndex]);
-        displayCardQuestion();
+        if (mShowingAnswer){
+            displayCardQuestion();
+            displayCardAnswer();
+        } else {
+            displayCardQuestion();
+        }
         showBackIcon();
     }
 
@@ -76,6 +87,14 @@ public class Previewer extends AbstractFlashcardViewer {
     }
 
 
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putInt("index", mIndex);
+        outState.putBoolean("showingAnswer", mShowingAnswer);
+    }
+
+
     @Override
     protected void displayCardQuestion() {
         super.displayCardQuestion();