diff --git a/app/src/main/java/io/pslab/fragment/ESPFragment.java b/app/src/main/java/io/pslab/fragment/ESPFragment.java
index 3c13b32..7ddc40a 100644
--- a/app/src/main/java/io/pslab/fragment/ESPFragment.java
+++ b/app/src/main/java/io/pslab/fragment/ESPFragment.java
@@ -106,7 +106,7 @@ public class ESPFragment extends DialogFragment {
                     ScienceLabCommon.setEspBaseIP(espIPAddress);
                 }
                 result = response.body().string();
-            } catch (IOException e) {
+            } catch (IOException | IllegalArgumentException e) {
                 e.printStackTrace();
             }
             return result;
