diff --git a/app/src/main/java/com/cohenadair/anglerslog/locations/ManageFishingSpotFragment.java b/app/src/main/java/com/cohenadair/anglerslog/locations/ManageFishingSpotFragment.java
index b7298fe..81f2ffa 100644
--- a/app/src/main/java/com/cohenadair/anglerslog/locations/ManageFishingSpotFragment.java
+++ b/app/src/main/java/com/cohenadair/anglerslog/locations/ManageFishingSpotFragment.java
@@ -124,15 +124,17 @@ public class ManageFishingSpotFragment extends ManageContentFragment {
         FishingSpot fishingSpot = getNewFishingSpot();
         fishingSpot.setName(mNameView.getInputText());
 
-        if (mMap != null) {
-            fishingSpot.setLatitude(mMap.getCameraPosition().target.latitude);
-            fishingSpot.setLongitude(mMap.getCameraPosition().target.longitude);
-        } else {
-            // in case Google Play Services isn't available, fishing spots can still be added and updated later
-            fishingSpot.setLatitude(0.0);
-            fishingSpot.setLongitude(0.0);
+        Double lat = getCoordinateInput(mLatitudeView);
+        Double lng = getCoordinateInput(mLongitudeView);
+
+        if (lat == null || lng == null || !mMapFragment.isValid(new LatLng(lat, lng))) {
+            AlertUtils.showError(getActivity(), R.string.msg_invalid_coordinate);
+            return false;
         }
 
+        fishingSpot.setLatitude(lat);
+        fishingSpot.setLongitude(lng);
+
         // name
         if (fishingSpot.isNameNull()) {
             AlertUtils.showError(getActivity(), R.string.error_name);
@@ -240,52 +242,44 @@ public class ManageFishingSpotFragment extends ManageContentFragment {
         refreshButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                if (mLatitudeView.getInputText() == null || mLongitudeView.getInputText() == null)
-                    return;
-
-                double lat = validateCoordinateInput(mLatitudeView);
-                double lng = validateCoordinateInput(mLongitudeView);
+                Double lat = getCoordinateInput(mLatitudeView);
+                Double lng = getCoordinateInput(mLongitudeView);
 
-                if (lat != -1 && lng != -1) {
+                if (lat != null && lng != null) {
                     LatLng latLng = new LatLng(lat, lng);
 
                     if (mMapFragment.isValid(latLng)) {
                         mMapFragment.updateCameraNoZoom(latLng);
+                    } else {
+                        AlertUtils.showError(getActivity(), R.string.msg_invalid_coordinate);
                     }
+                } else {
+                    AlertUtils.showError(getActivity(), R.string.msg_invalid_coordinate);
                 }
             }
         });
     }
 
     private void updateCoordinateViews(LatLng coordinates) {
-        mLatitudeView.setInputText(String.format(getResources().getConfiguration().locale, "%.6f", coordinates.latitude));
-        mLongitudeView.setInputText(String.format(getResources().getConfiguration().locale, "%.6f", coordinates.longitude));
+        mLatitudeView.setInputText(String.format(java.util.Locale.US, "%.6f", coordinates.latitude));
+        mLongitudeView.setInputText(String.format(java.util.Locale.US, "%.6f", coordinates.longitude));
     }
 
-    /**
-     * Validates the given {@link InputTextView}. If the input is invalid a Toast will be displayed
-     * to the user and the previously entered character will be removed.
-     *
-     * @param input The {@link InputTextView} to validate.
-     * @return The text of the input as a double.
-     */
-    private double validateCoordinateInput(InputTextView input) {
+    private Double getCoordinateInput(InputTextView input) {
         String inputText = input.getInputText();
 
-        // do not validate if the user is entering a negative coordinate
-        if (inputText.length() == 1 && inputText.charAt(0) == '-')
-            return -1;
+        if (inputText == null)
+            return 0.0;
 
         try {
-            inputText = inputText.replace(',', '.'); // some countries/keyboard use ',' in place of '.'
-            return (double) Utils.asFloat(inputText, 0);
+            inputText = inputText.replace(',', '.');
+            if (inputText.isEmpty() || (inputText.length() == 1 && inputText.equals(".")))
+                return 0.0;
+
+            return Double.parseDouble(inputText);
         } catch (NumberFormatException e) {
-            e.printStackTrace();
-            Utils.showToast(getContext(), R.string.msg_invalid_coordinate);
-            input.pressBackspace();
+            return null;
         }
-
-        return -1;
     }
 
     private void initMapFragment() {
diff --git a/app/src/main/java/com/cohenadair/anglerslog/model/user_defines/FishingSpot.java b/app/src/main/java/com/cohenadair/anglerslog/model/user_defines/FishingSpot.java
index bdd97d6..a3e042e 100644
--- a/app/src/main/java/com/cohenadair/anglerslog/model/user_defines/FishingSpot.java
+++ b/app/src/main/java/com/cohenadair/anglerslog/model/user_defines/FishingSpot.java
@@ -113,7 +113,7 @@ public class FishingSpot extends UserDefineObject {
      * @return A String of the FishingSpot's coordinates.
      */
     public String getCoordinatesAsString(String lat, String lng) {
-        return String.format(lat + ": %.6f, " + lng + ": %.6f", mLatitude, mLongitude);
+        return String.format(java.util.Locale.US, lat + ": %.6f, " + lng + ": %.6f", mLatitude, mLongitude);
     }
 
     /**
@@ -122,7 +122,7 @@ public class FishingSpot extends UserDefineObject {
      * @see #getCoordinatesAsString(String, String)
      */
     public String getCoordinatesAsString() {
-        return String.format("%.6f, %.6f", mLatitude, mLongitude);
+        return String.format(java.util.Locale.US, "%.6f, %.6f", mLatitude, mLongitude);
     }
 
     /**
