diff --git a/app/src/main/java/org/shadowice/flocke/andotp/Activities/IntroScreenActivity.java b/app/src/main/java/org/shadowice/flocke/andotp/Activities/IntroScreenActivity.java
index 3e7d383..c8344a5 100644
--- a/app/src/main/java/org/shadowice/flocke/andotp/Activities/IntroScreenActivity.java
+++ b/app/src/main/java/org/shadowice/flocke/andotp/Activities/IntroScreenActivity.java
@@ -229,7 +229,7 @@ public class IntroScreenActivity extends IntroActivity {
     }
 
     public static class AuthenticationFragment extends SlideFragment {
-        private Constants.EncryptionType encryptionType = Constants.EncryptionType.KEYSTORE;
+        private Constants.EncryptionType encryptionType = Constants.EncryptionType.PASSWORD;
 
         private int slidePos = -1;
 
@@ -261,6 +261,10 @@ public class IntroScreenActivity extends IntroActivity {
             if (desc != null) {
                 if (encryptionType == Constants.EncryptionType.KEYSTORE) {
                     desc.setText(R.string.intro_slide3_desc_keystore);
+
+                    Constants.AuthMethod selectedMethod = selectionMapping.get(selection.getSelectedItemPosition());
+                    if (selectedMethod == Constants.AuthMethod.PASSWORD || selectedMethod == Constants.AuthMethod.PIN )
+                        selection.setSelection(selectionMapping.indexOfValue(Constants.AuthMethod.NONE));
                 } else if (encryptionType == Constants.EncryptionType.PASSWORD) {
                     desc.setText(R.string.intro_slide3_desc_password);
 
@@ -431,7 +435,13 @@ public class IntroScreenActivity extends IntroActivity {
             passwordInput.addTextChangedListener(textWatcher);
             passwordConfirm.addTextChangedListener(textWatcher);
 
-            selection.setSelection(selectionMapping.indexOfValue(Constants.AuthMethod.PASSWORD));
+            if (encryptionType == Constants.EncryptionType.KEYSTORE) {
+                desc.setText(R.string.intro_slide3_desc_keystore);
+                selection.setSelection(selectionMapping.indexOfValue(Constants.AuthMethod.NONE));
+            } else if (encryptionType == Constants.EncryptionType.PASSWORD) {
+                desc.setText(R.string.intro_slide3_desc_password);
+                selection.setSelection(selectionMapping.indexOfValue(Constants.AuthMethod.PASSWORD));
+            }
 
             return root;
         }
