diff --git a/AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java b/AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java
index a8dba17..e6f03dc 100644
--- a/AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java
@@ -43,6 +43,10 @@ public class Previewer extends AbstractFlashcardViewer {
 
         mCardList = getIntent().getLongArrayExtra("cardList");
         mIndex = getIntent().getIntExtra("index", -1);
+        if (savedInstanceState != null) {
+            mIndex = savedInstanceState.getInt("index", mIndex);
+            mShowingAnswer = savedInstanceState.getBoolean("mShowingAnswer", false);
+        }
         if (mCardList.length == 0 || mIndex < 0 || mIndex > mCardList.length - 1) {
             Timber.e("Previewer started with empty card list or invalid index");
             finishWithoutAnimation();
@@ -54,11 +58,22 @@ public class Previewer extends AbstractFlashcardViewer {
         startLoadingCollection();
     }
 
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putInt("index", mIndex);
+        outState.putBoolean("mShowingAnswer", mShowingAnswer);
+    }
+
     @Override
     protected void onCollectionLoaded(Collection col) {
         super.onCollectionLoaded(col);
         mCurrentCard = col.getCard(mCardList[mIndex]);
-        displayCardQuestion();
+        if (mShowingAnswer) {
+            displayCardAnswer();
+        } else {
+            displayCardQuestion();
+        }
         showBackIcon();
     }
 
