--- a/app/src/main/java/de/danoeh/antennapod/adapter/AllEpisodesRecycleAdapter.java
+++ b/app/src/main/java/de/danoeh/antennapod/adapter/AllEpisodesRecycleAdapter.java
@@ -106,6 +106,9 @@
     public void onBindViewHolder(final Holder holder, int position) {
         final FeedItem item = itemAccess.getItem(position);
         if (item == null) return;
+        holder.itemView.setAlpha(1.0f);
+        holder.itemView.setTranslationX(0f);
+        holder.itemView.setTranslationY(0f);
         holder.itemView.setOnLongClickListener(v -> {
             this.position = holder.getAdapterPosition();
             return false;
@@ -270,6 +273,8 @@
         @Override
         public void onItemClear() {
             itemView.setAlpha(1.0f);
+            itemView.setTranslationX(0f);
+            itemView.setTranslationY(0f);
         }
 
         public FeedItem getFeedItem() { return item; }
--- a/app/src/main/java/de/danoeh/antennapod/adapter/QueueRecyclerAdapter.java
+++ b/app/src/main/java/de/danoeh/antennapod/adapter/QueueRecyclerAdapter.java
@@ -200,10 +200,15 @@
         @Override
         public void onItemClear() {
             itemView.setAlpha(1.0f);
+            itemView.setTranslationX(0f);
+            itemView.setTranslationY(0f);
         }
 
         public void bind(FeedItem item) {
             this.item = item;
+            itemView.setAlpha(1.0f);
+            itemView.setTranslationX(0f);
+            itemView.setTranslationY(0f);
             if(locked) {
                 dragHandle.setVisibility(View.GONE);
             } else {
--- a/app/src/main/java/de/danoeh/antennapod/fragment/AllEpisodesFragment.java
+++ b/app/src/main/java/de/danoeh/antennapod/fragment/AllEpisodesFragment.java
@@ -343,7 +343,6 @@
             MainActivity mainActivity = (MainActivity) getActivity();
             listAdapter = new AllEpisodesRecycleAdapter(mainActivity, itemAccess,
                     new DefaultActionButtonCallback(mainActivity), showOnlyNewEpisodes());
-            listAdapter.setHasStableIds(true);
             recyclerView.setAdapter(listAdapter);
         }
         listAdapter.notifyDataSetChanged();
--- a/app/src/main/java/de/danoeh/antennapod/fragment/QueueFragment.java
+++ b/app/src/main/java/de/danoeh/antennapod/fragment/QueueFragment.java
@@ -507,7 +507,6 @@
             MainActivity activity = (MainActivity) getActivity();
             recyclerAdapter = new QueueRecyclerAdapter(activity, itemAccess,
                 new DefaultActionButtonCallback(activity), itemTouchHelper);
-            recyclerAdapter.setHasStableIds(true);
             recyclerView.setAdapter(recyclerAdapter);
         }
         if(queue == null || queue.size() == 0) {