--- a/GPSTest/src/main/java/com/android/gpstest/GpsStatusFragment.java
+++ b/GPSTest/src/main/java/com/android/gpstest/GpsStatusFragment.java
@@ -306,6 +306,9 @@
                 mSbasStatus.clear();
                 mGnssAdapter.notifyDataSetChanged();
                 mSbasAdapter.notifyDataSetChanged();
+                filterGroup.setVisibility(View.GONE);
+                filterTextView.setText("");
+                mNumSats.setTypeface(null, Typeface.NORMAL);
             }
             mNavigating = navigating;
         }
@@ -443,9 +446,11 @@
 
     @SuppressLint("NewApi")
     public void gpsStart() {
+        setStarted(true);
     }
 
     public void gpsStop() {
+        setStarted(false);
     }
 
     public void onLocationChanged(Location location) {